igvm/hypervisor.py

```diff
@@ -215,6 +190,40 @@ def check_vm(self, vm, offline):
                     .format(*hw_pair)
                 )
 
+        # HV in supported state?
+        if self.dataset_obj['state'] not in ['online', 'online_reserved']:
```

Comment:
I think this decision should be done when filtering for the hypervisors to check in the first place. This would make it impossible to change the hypervisor selection from the commandline with arguments such as --ignore-reserved.

See: https://github.com/innogames/igvm/blob/master/igvm/commands.py#L312

Reply:
Interesting observation. But this code is then even more wrong: a HV could change its state whille it's being searched for. We should check here for exactly requested state.

Reply:
lets wait for feedback from @TuxPowered42 but addresse this in another pr since the logic is the same as before, just in another place